Core/Core/Kits/Applications.cs

```diff
@@ -53,45 +56,46 @@ public string GetVersion(HostAppVersion version)
-  public static HostApplication TeklaStructures = new("Tekla Structures", "teklastructures");
-  public static HostApplication Dxf = new("DXF Converter", "dxf");
-  public static HostApplication Excel = new("Excel", "excel");
-  public static HostApplication Unreal = new("Unreal", "unreal");
-  public static HostApplication PowerBI = new("Power BI", "powerbi");
-  public static HostApplication Blender = new("Blender", "blender");
-  public static HostApplication QGIS = new("QGIS", "qgis");
-  public static HostApplication ArcGIS = new("ArcGIS", "arcgis");
-  public static HostApplication SketchUp = new("SketchUp", "sketchup");
-  public static HostApplication Archicad = new("Archicad", "archicad");
-  public static HostApplication TopSolid = new("TopSolid", "topsolid");
-  public static HostApplication Python = new("Python", "python");
-  public static HostApplication NET = new(".NET", "net");
-  public static HostApplication Navisworks = new("Navisworks", "navisworks");
-  public static HostApplication AdvanceSteel = new("Advance Steel", "advancesteel");
-  public static HostApplication Other = new("Other", "other");
+  public static readonly HostApplication Rhino = new("Rhino", "rhino"),
+    Grasshopper = new("Grasshopper", "grasshopper"),
+    Revit = new("Revit", "revit"),
+    Dynamo = new("Dynamo", "dynamo"),
+    Unity = new("Unity", "unity"),
+    GSA = new("GSA", "gsa"),
+    Civil = new("Civil 3D", "civil3d"),
+    AutoCAD = new("AutoCAD", "autocad"),
+    MicroStation = new("MicroStation", "microstation"),
+    OpenRoads = new("OpenRoads", "openroads"),
+    OpenRail = new("OpenRail", "openrail"),
+    OpenBuildings = new("OpenBuildings", "openbuildings"),
+    ETABS = new("ETABS", "etabs"),
+    SAP2000 = new("SAP2000", "sap2000"),
+    CSiBridge = new("CSiBridge", "csibridge"),
+    SAFE = new("SAFE", "safe"),
+    TeklaStructures = new("Tekla Structures", "teklastructures"),
+    Dxf = new("DXF Converter", "dxf"),
+    Excel = new("Excel", "excel"),
+    Unreal = new("Unreal", "unreal"),
+    PowerBI = new("Power BI", "powerbi"),
+    Blender = new("Blender", "blender"),
+    QGIS = new("QGIS", "qgis"),
+    ArcGIS = new("ArcGIS", "arcgis"),
+    SketchUp = new("SketchUp", "sketchup"),
+    Archicad = new("Archicad", "archicad"),
+    TopSolid = new("TopSolid", "topsolid"),
+    Python = new("Python", "python"),
+    NET = new(".NET", "net"),
+    Navisworks = new("Navisworks", "navisworks"),
+    AdvanceSteel = new("Advance Steel", "advancesteel"),
+    Other = new("Other", "other");
 
   /// <summary>
   /// Gets a HostApplication form a string. It could be the versioned name or a string coming from a process running.
   /// </summary>
   /// <param name="appname">String with the name of the app</param>
   /// <returns></returns>
-  public static HostApplication GetHostAppFromString(string appname)
+  [Pure]
+  public static HostApplication GetHostAppFromString(string? appname)
```

Comment:
could we not create a static dinctionary here of the HostApplication instances and turn this into a bit of linq? I think that is cleaner and a bit more extensible?

Reply:
This is all pre-existing. I don't like how it is, a dictionary would be so much nicer. But for now, I didn't want to refactor any more than necessary.